File: sample/sample-android/src/main/java/com/example/sampleandroid/view/model/DeeplinkSubScreen.kt
package com.example.sampleandroid.view.model

import androidx.compose.foundation.layout.Column
import androidx.compose.runtime.Composable
import com.example.sampleandroid.view.widget.SampleTextField
import kim.jeonghyeon.androidlibrary.compose.Screen
import kim.jeonghyeon.androidlibrary.compose.widget.Button
import kim.jeonghyeon.sample.viewmodel.DeeplinkSubViewModel

@Composable
fun DeeplinkSubScreen(model: DeeplinkSubViewModel) {
    Screen(model) {
        Column {
            SampleTextField("Input value", model.result)
            Button("OK") { model.onClickOk() }
        }
    }
}

// TODO reactive way.
//@Composable
//fun DeeplinkSubScreen2(model: DeeplinkSubViewModel2) {
//    Screen(model) {
//        Column {
//            SampleTextField("Input value", model.result)
//            Button("OK", model.click)
//        }
//    }
//}

File: sample/sample-native/kotlinIOS/screen/model/DeeplinkSubScreen.swift
//
//  SigninScreen.swift
//  kotlinIOS
//
//  Created by hyun kim on 04/09/20.
//  Copyright © 2020 hyun kim. All rights reserved.
//

import SwiftUI
import sample_base

func DeeplinkSubScreen(_ model: DeeplinkSubViewModel) -> some View {
    Screen(model) {
        Column {
            SampleTextField("Input value", model.result)
            Button("OK") { model.onClickOk() }
        }
    }
}
